### PR TITLE
raftstore: prettify snapshot build flow

### PR DESCRIPTION
**Pull request by Connor1996**
Signed-off-by: Connor1996 <zbk602423539@gmail.com>

### What is changed and how it works?

Issue Number: Ref #12876 

What's Changed:

```commit-message
prettify snapshot build flow
```

### Check List 

Tests 

- No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/raftstore/src/store/peer_storage.rs [hunk 1]
```diff
@@ -1003,72 +1000,57 @@ where
...
     );
 
-    let msg = kv_snap
+    let apply_state: RaftApplyState = kv_snap
         .get_msg_cf(CF_RAFT, &keys::apply_state_key(region_id))
-        .map_err(into_other::<_, raft::Error>)?;
-    let apply_state: RaftApplyState = match msg {
-        None => {
-            return Err(storage_error(format!(
+        .and_then(|res| match res {
```
Comment (BusyJay) [inline components/raftstore/src/store/peer_storage.rs:1005]: You mean `res.ok_or_else`?

File: components/raftstore/src/store/peer_storage.rs [hunk 2]
```diff
@@ -1003,72 +1000,57 @@ where
...
+            None => Err(box_err!(
```
Comment (BusyJay) [inline components/raftstore/src/store/peer_storage.rs:1006]: This is too implicit. What will the exact error be returned?

Comment (Connor1996) [reply components/raftstore/src/store/peer_storage.rs:1006]: I don't know, probably `Error::Other`. L1026 does the same way.

Comment (Connor1996) [reply components/raftstore/src/store/peer_storage.rs]: refine it

**Review by BusyJay — APPROVED** [review]

**Review by ethercflow — APPROVED** [review]
LGTM

**Pull request merged**